spec/gtfs_parser_spec.rb

```diff
@@ -14,5 +13,31 @@
         expect(result[%w[31 31 0 Sunderland]]).is_a? Array
       end
     end
+    context 'with a trip ending at a stop and another beginning at the stop' do
+      let(:result) { find_departures(%w[stop_id]) }
+      it 'does not throw out the departure at the start of the trip' do
+        expect_any_instance_of(GtfsParser)
+          .to receive(:find_trips_operating_today).with('stop_id')
+          .and_return('trip_id1' => :route_data1, 'trip_id2' => :route_data2)
+        row0 = { 'trip_id' => 'trip_id1', 'stop_id' => 'something else' }
+        # row1 marks the end of a trip. Since the trip ends at the stop, it is
```

Comment:
I'm glad they help :-)

Hopefully they'll help me too the next time I need to come back and work with this code.